### PR TITLE
Auto screenshot improvements

### PR DESCRIPTION
**Pull request by patrick-stephens**
Primarily to fix the failures on the action which we could not test until it was merged.
This also uploads the screenshots taken to the action run.

### CODE REVIEW

**Comment by patrick-stephens** [comment]
Need to confirm if this just does changed files or not, I think it will do all and we need to tweak the `generate-tools/screenshots.sh` to run without `all` for actions.

It also now runs Hadolint across all Dockerfiles it finds, ignoring the Tern ones and any dockerignore files. Assumption is a Dockerfile starts with that `Dockerfile*` prefix.

**Comment by patrick-stephens** [comment]
See https://github.com/couchbaselabs/observability/actions/runs/1425881683 for an example run - the screenshots are archived with the workflow run currently.

File: tools/hadolint.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -eu
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+# Ensure we always have latest version
+docker pull hadolint/hadolint
+
+# Find all Dockerfiles assuming a certain naming convention
+exitCode=0
```
Comment (markspolakovs) [inline tools/hadolint.sh:21]: NITPICK: `exit_code` for consistency

File: microlith/Dockerfile.rhel [hunk 2]
```diff
@@ -104,7 +110,7 @@ ENV PATH="/usr/share/grafana/bin:$PATH" \
 
 WORKDIR $GF_PATHS_HOME
 
-# hadolint ignore=DL3059,DL4006,SC2155
+# hadolint ignore=DL3059,DL4006,SC2155,DL3046
```
Comment (markspolakovs) [inline microlith/Dockerfile.rhel:113]: NITPICK: possibly a comment explaining what these mean?

**Comment by patrick-stephens** [comment]
This should also resolve https://issues.couchbase.com/browse/CMOS-132

**Review by markspolakovs — APPROVED** [review]
For reference, comments with NITPICK you are fully free to ignore if you wish.

**Pull request merged**